### PR TITLE
make plot and table interactions programmable #2888

### PR DESCRIPTION
**Pull request by mnavasiolava**
Please find notebook with examples in our tutorial ("Plot actions").
Looking forward to your comments and remarks

### CODE REVIEW

**Comment by scottdraves** [comment]
thanks this looks great, need to do some tests...

**Comment by scottdraves** [comment]
kick jenkins

**Comment by scottdraves** [comment]
:+1:

File: plugin/jvm/src/main/java/com/twosigma/beaker/chart/serializer/ObservableChartSerializer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.twosigma.beaker.chart.ObservableChart;
+import com.twosigma.beaker.chart.actions.ChartObjectManager;
+import com.twosigma.beaker.jvm.updater.UpdateManager;
+import org.codehaus.jackson.JsonGenerator;
+import org.codehaus.jackson.map.JsonSerializer;
+
+import java.io.IOException;
+
+@Singleton
+public abstract class ObservableChartSerializer<T extends ObservableChart> extends JsonSerializer<T> {
```
Comment (tomca32) [inline plugin/jvm/src/main/java/com/twosigma/beaker/chart/serializer/ObservableChartSerializer.java:15]: @mnavasiolava Why is this both a Singleton and abstract? I don't know much about Guice so I might be missing something but I don't see how can an abstract class be a Singleton, or be instantiated at all.

Comment (mnavasiolava) [inline plugin/jvm/src/main/java/com/twosigma/beaker/chart/serializer/ObservableChartSerializer.java:15]: @tomca32 you're right, singleton annotation shouldn't be there.

**Pull request merged**